core/controllers/admin.py

```diff
@@ -656,6 +675,390 @@ def _generate_dummy_explorations(
+            'placeholder': {
+                'value': {
+                    'content_id': 'ca_placeholder_0',
+                    'unicode_str': ''
+                }
+            },
+            'rows': {'value': 1}
+        })
+
+        state.update_next_content_id_index(1)
+        state.update_linked_skill_id(None)
+        state.update_content(
+            state_domain.SubtitledHtml('1', 'Content 1'))
+        recorded_voiceovers = state_domain.RecordedVoiceovers({})
+        written_translations = state_domain.WrittenTranslations({})
+        recorded_voiceovers.add_content_id_for_voiceover(
+            'ca_placeholder_0')
+        recorded_voiceovers.add_content_id_for_voiceover('1')
+        recorded_voiceovers.add_content_id_for_voiceover(
+            'default_outcome')
+        written_translations.add_content_id_for_translation(
+            'ca_placeholder_0')
+        written_translations.add_content_id_for_translation('1')
+        written_translations.add_content_id_for_translation(
+            'default_outcome')
+
+        state.update_recorded_voiceovers(recorded_voiceovers)
+        state.update_written_translations(written_translations)
+        solution = state_domain.Solution(
+            'TextInput', False, 'Solution', state_domain.SubtitledHtml(
+                'solution', '<p>This is a solution.</p>'))
+        hints_list = [
+            state_domain.Hint(
+                state_domain.SubtitledHtml(
+                    'hint_1', '<p>This is a hint.</p>')
+            )
+        ]
+
+        state.update_interaction_solution(solution)
+        state.update_interaction_hints(hints_list)
+        state.update_interaction_default_outcome(
+            state_domain.Outcome(
+                dest, state_domain.SubtitledHtml(
+                    'feedback_id', '<p>Dummy Feedback</p>'),
+                True, [], None, None
+            )
+        )
+
+    def _generate_dummy_opportunities(
+            self, num_dummy_ops_to_generate,
+            num_dummy_interactions):
+        """Generates and publishes the given number of dummy opportunities.
+
+        Args:
+            num_dummy_ops_to_generate: int. Count of dummy opportunities to
+                be generated.
+            num_dummy_interactions: int. Count of dummy
+                interactions to be generate.
+
+        Raises:
+            Exception. Environment is not DEVMODE.
+        """
+
+        if constants.DEV_MODE:
+            if story_services.does_story_exist_with_url_fragment(
+                'help-jamie-win-arcade'):
+                self._generate_dummy_opportunities_with_existing_topic(
+                    num_dummy_ops_to_generate,
+                    num_dummy_interactions,
+                    'help-jamie-win-arcade')
+            else:
+                self._generate_dummy_opportunities_without_existing_topic(
+                    num_dummy_ops_to_generate,
+                    num_dummy_interactions)
```

Comment:
There needs to be seperate code paths because if no topic exists, we need to create a new topic and story. 

If a topic exists we need to add new story nodes to an existing story, in order to update an existing story you need to create a `changeList = [StoryChange ..]` to confirm changes to the story as well